lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -139,6 +147,31 @@ def get_request_id(headers, env)
           headers ||= {}
           headers['X-Request-Id'] || headers['X-Request-ID'] || env['HTTP_X_REQUEST_ID']
         end
+
+        private
+
+        REQUEST_SPAN_DEPRECATION_WARNING = %(
+          Use of :datadog_rack_request_span in the Rack env is DEPRECATED.
+          If you need the Rack request span, try using `Datadog.tracer.active_span`.
+          This key will no longer be available in future versions.).freeze
+
+        def add_deprecation_warnings(env)
+          env.instance_eval do
+            def [](key)
+              if key == :datadog_rack_request_span
+                Datadog::Tracer.log.warn(REQUEST_SPAN_DEPRECATION_WARNING)
+              end
+              super
+            end
+
+            def []=(key, value)
```

Comment:
the setter is required as a deprecation warning? we're not setting `: datadog_rack_request_span` anymore and if users do, it's not something we should not interfere I think

Reply:
Was doing this out of abundance of caution, in case someone was overriding this value. Seems very unlikely, but possible.